python_modules/dagster/dagster/_model/__init__.py

```diff
@@ -15,3 +17,10 @@ def __init__(self, **data: Any) -> None:
     class Config:
         extra = "forbid"
         frozen = True
+
+    def model_copy(self, *, update: Optional[Mapping[str, Any]] = None, deep: bool = False) -> Self:
+        if pydantic.__version__ >= "2":
+            func = getattr(BaseModel, "model_copy")
+        else:
+            func = getattr(BaseModel, "copy")
+        return func(self, update=update, deep=deep)
```

Comment:
would not a `super()` call be a little more obvious? Would need to `type: ignore` the v1 codepath

Reply:
👍 - switched to super()